src/test/kotlin/com/edd/memegrid/web/RouterTest.kt

```diff
@@ -0,0 +1,285 @@
+        }
+
+        assertThat(resultMeme, equalTo(meme))
+    }
+
+    @Test
+    fun `should return 200 and deleted meme given delete request for existing meme`() {
+        val memeId: Long = 1
+        val meme = Meme(memeId, "Good meme", "url")
+        every { mockMemeManager.getMeme(memeId) } returns meme
+        every { mockMemeManager.deleteMeme(meme) } returns 1
+
+        val resultMeme = given()
+                    .port(PORT)
+                    .accept(ContentType.JSON)
+                .delete("/api/memes/{id}", memeId)
+                .then()
+                    .statusCode(200)
+                .extract()
+                    .`as`(Meme::class.java)
+
+        assertThat(resultMeme, equalTo(meme))
+    }
+
+    @Test
+    fun `should return 200 and memes given get request for page`() {
+
+        val memesOnPage = listOf(Meme(1, "Good meme", "/meme/url"),
+                Meme(2, "Better meme", "/better/meme/url"))
+
+        every { mockMemeManager.getMemes(any()) } returns memesOnPage
+
+        val page: Long = 1
+        val resultJsonNode = given()
+            .port(PORT)
+            .accept(ContentType.JSON)
+        .get("/api/memes/page/{page}", page)
+        .then()
+            .statusCode(200)
+        .extract()
+                .`as`(JsonNode::class.java)
+
+        assertThat(resultJsonNode.toMemeList(), equalTo(memesOnPage))
+    }
+
+    @Test
+    fun `should return 404 with json body given get request that accepts json to invalid path`() {
+        Given {
+            port(PORT)
+            accept(ContentType.JSON)
+        } When {
+            get("/invalid")
+        } Then {
+            statusCode(STATUS_CODE_NOT_FOUND)
+            body(containsString("Path does not exist"))
+        }
+    }
+
+    @Test
+    fun `should return 404 with html body given get request that accepts html to invalid path`() {
+        val htmlResponseBody = "Not found HTML"
+        every { mockTemplateReader.read(any(), any()) } returns htmlResponseBody
+
+        Given {
+            port(PORT)
+            accept(ContentType.HTML)
+        } When {
+            get("/invalid")
+        } Then {
+            statusCode(STATUS_CODE_NOT_FOUND)
+            body(containsString(htmlResponseBody))
+        }
+    }
+
+    @Test
+    fun `should return 500 when saving a new meme fails`() {
+        val meme = Meme(1, "Good meme", "url")
+        every { mockImageValidator.isValid(any()) } returns true
+        every { mockMemeManager.getMeme(any<String>()) } returns null
+        every { mockMemeManager.saveMeme(any(), any()) } throws Exception()
+
+        Given {
+            port(PORT)
+            accept(ContentType.JSON)
+            body(meme)
+        } When {
+            post("/api/memes")
+        } Then {
+            statusCode(STATUS_CODE_SERVER_ERROR)
+        }
+    }
+
+    private fun JsonNode.toMemeList() = jacksonObjectMapper().readValue<List<Meme>>(this.toPrettyString())
```

Comment:
I'd suggest to place `jacksonObjectMapper()` inside the `companion object`, assign it to a `val` and reference it here. As `jacksonObjectMapper` creates a new instance of `ObjectMapper` each time it is called:
```kotlin
fun jacksonObjectMapper(): ObjectMapper = ObjectMapper().registerKotlinModule()
```

Reply:
Moved to companion object